src/main/java/betterquesting/api2/cache/QuestCache.java

```diff
@@ -128,17 +121,21 @@ public synchronized void updateCache(EntityPlayer player) {
                 tmpVisible.add(entry.getID());
             }
         }
+        int[] tmpActiveArr = tmpActive.toIntArray();
+        if (!Arrays.equals(getActiveQuests(), tmpActiveArr)) {
+            QuestingAPI.getAPI(ApiReference.QUEST_DB).invalidateBulkCache(uuid);
+        }
 
         visibleQuests.clear();
-        visibleQuests.addAll(tmpVisible);
-
         activeQuests.clear();
-        activeQuests.addAll(tmpActive);
-
         resetSchedule.clear();
+        autoClaims.clear();
+
+        // Copy the temp sorted sets to array lists for better locality
+        visibleQuests.addAll(tmpVisible);
+        activeQuests.addElements(0, tmpActiveArr);
```

Comment:
it seems like `activeQuests.addAll(tmpActive);` would still work - why switch to `activeQuests.addElements(0, tmpActiveArr);` here?